### PR TITLE
Core/Scripts/DB: remove hardcoded text from 'npcs_dithers_and_arbington'

### PR DESCRIPTION
**Pull request by ghost**
- added enum section instead of "magic numbers" in script
- replaced core text with pointers to DB text
- created DB gossip_menu_option entries for both NPCs
- zone_western_plaguelands.cpp is now cleaned up

### CODE REVIEW

**Comment by tkrokli** [comment]
It looks like @runningnak3d is the original author of hardcoded text according to Git Blame (file history).

File: src/server/scripts/EasternKingdoms/zone_western_plaguelands.cpp [hunk 1]
```diff
@@ -41,12 +41,17 @@ EndContentData */
...
 
-#define GOSSIP_HDA1 "What does the Felstone Field Cauldron need?"
-#define GOSSIP_HDA2 "What does the Dalson's Tears Cauldron need?"
-#define GOSSIP_HDA3 "What does the Writhing Haunt Cauldron need?"
-#define GOSSIP_HDA4 "What does the Gahrron's Withering Cauldron need?"
-
-#define GOSSIP_SDA1 "Thanks, i need a Vitreous Focuser"
+enum DithersAndArbington
+{
+    OSSEOUS_AGITATORS       = 3980,
```
Comment (DDuarte) [inline src/server/scripts/EasternKingdoms/zone_western_plaguelands.cpp]: prefix gossip ids with GOSSIP_ or something similar

Comment (tkrokli) [inline src/server/scripts/EasternKingdoms/zone_western_plaguelands.cpp]: :+1: OK, good point. I guess I haven't read the guidelines properly.

**Comment by jackpoz** [comment]
> Edit: This PR has been tested both from the horde and alliance side and works as intended.

Nice to see contributors test the changes and stating it in the PR.

**Comment by tkrokli** [comment]
Thanks, I should probably make it a rule to state that in the commit message of my future PRs.

File: sql/updates/world/2016_01_15_05_world_335.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+(3223,1,0,"What's needed for the cauldron at Felstone Field?",6516,3,131,0,0,0,0,'',0),
+(3223,2,0,"What's needed for the cauldron at Dalson's Tears?",6517,3,131,0,0,0,0,'',0),
+(3223,3,0,"What's needed for the cauldron at the Writhing Haunt?",6518,3,131,0,0,0,0,'',0),
+(3223,4,0,"What's needed for the cauldron at Gahrron's Withering?",6519,3,131,0,0,0,0,'',0),
+(3223,5,0,'I need a Vitreous Focuser.',6531,3,131,0,0,0,0,'',0),
+(3228,1,0,"What's needed for the cauldron at Felstone Field?",6516,3,131,0,0,0,0,'',0),
+(3228,2,0,"What's needed for the cauldron at Dalson's Tears?",6517,3,131,0,0,0,0,'',0),
+(3228,3,0,"What's needed for the cauldron at the Writhing Haunt?",6518,3,131,0,0,0,0,'',0),
+(3228,4,0,"What's needed for the cauldron at Gahrron's Withering?",6519,3,131,0,0,0,0,'',0),
+(3228,5,0,"I need a Vitreous Focuser.",6531,3,131,0,0,0,0,'',0);
```
Comment (joschiwald) [inline sql/updates/world/2016_01_15_05_world_335.sql]: i'm sure these gossip ids are incorrect, that's not the way how blizz creates gossips (missing submenus)

Comment (tkrokli) [inline sql/updates/world/2016_01_15_05_world_335.sql]: You say "(missing submenus)", I don't get it. I can click on those texts from https://github.com/TrinityCore/TrinityCore/pull/16276/files#diff-a0f88ddfc89e397f162f6ef2a289da41R103 (+3) and get displayed the lines from `npc_text`.`ID` 3980,3981,3982 and 3983 with the option to click "I need a Vitreous Focuser." (https://github.com/TrinityCore/TrinityCore/pull/16276/files#diff-bc94fdb44cfe02ee74501deb5a08189bR13). From what I could see, the gossip menus behave the same, before and after changing the script, as long as the player has completed the required quest chains.

Comment (joschiwald) [inline sql/updates/world/2016_01_15_05_world_335.sql]: i'm to 99% sure, options 1-4 and 5 are different menus

Comment (tkrokli) [inline sql/updates/world/2016_01_15_05_world_335.sql]: OK, now I get what you mean. So 1-4 are OK, but not 5 within the same set as 1-4.

**Pull request merged**